tests/network/expose.go

```diff
@@ -133,9 +115,22 @@ var _ = SIGDescribe("[rfe_id:253][crit:medium][vendor:cnv-qe@redhat.com][level:c
 		return vmiExposeArgs
 	}
 
-	createAndWaitForJobToSucceed := func(helloWorldJobCreator func(host, port, namespace string) *batchv1.Job, namespace, ip, port, viaMessage string) error {
+	createAndWaitForJobToSucceed := func(proto, namespace, ip, port, viaMessage string) error {
+		var job *batchv1.Job
+
 		By(fmt.Sprintf("Starting a job which tries to reach the VMI via the %s", viaMessage))
-		job := helloWorldJobCreator(ip, port, namespace)
+
+		if proto == "UDP" {
+			job = tests.NewHelloWorldJobUDP(ip, port)
+		} else if proto == "HTTP" {
+			job = tests.NewHelloWorldJobHTTP(ip, port)
+		} else if proto == "TCP" {
+			job = tests.NewHelloWorldJobTCP(ip, port)
```

Comment:
> Instead of a proto string, I would expect to see here an object or a function pointer for tests.NewHelloWorldJob*.

I don't like the name and location of tests.NewHelloWorldJob*, but I can live with your suggestion if it grants me your lgtm.